Source/Core/VideoCommon/TmemBase.cpp

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <array>
+
+#include "VideoCommon/BPMemory.h"
+#include "VideoCommon/TmemBase.h"
+
+namespace TmemBase
+{
+// On TMEM configuration changed:
+// 1. invalidate stage.
+
+struct texture_unit_state
+{
+  enum State
+  {
+    INVALID, // Configuration has changed
```

Comment:
Thoughts on putting the comment in the line before rather than at the end of the line? At least Visual Studio shows those in Intellisense if they're in the line before.

Reply:
Sure, why not.